File: go/count-number-of-pairs-with-absolute-difference-k.go
func countKDifference(nums []int, k int) int {
	previous := make(map[int]int)
	count := 0
	for i := 0; i < len(nums); i++ {
		num := nums[i]
		count += previous[num-k]
		count += previous[num+k]
		previous[num] += 1
	}
	return count
}

File: python/count-number-of-pairs-with-absolute-difference-k.py
class Solution:
    def countKDifference(self, nums: List[int], k: int) -> int:
        previous = defaultdict(lambda: 0)
        pair_count = 0
        for num in nums:
            pair_count += previous[num - k]
            pair_count += previous[num + k]
            previous[num] += 1
        return pair_count
